gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContents.java

```diff
@@ -0,0 +1,379 @@
+        result -> identifiedResult.getContentValues().putAll(result.getContentValues()));
+    return identifiedResult;
+  }
+
+  protected Map<String, ContentBloomFilter> computeLiveContents(
+      Instant cutOffTimestamp, Reference reference, Instant droppedRefTime) {
+    try (NessieApiV1 api = getApi(config)) {
+      boolean isRefDroppedAfterCutoffTimeStamp =
+          droppedRefTime == null || droppedRefTime.compareTo(cutOffTimestamp) >= 0;
+      return walkLiveCommitsInReference(
+          api,
+          reference,
+          isRefDroppedAfterCutoffTimeStamp,
+          commitMeta ->
+              // If the commit time is newer than (think: greater than or equal to) cutoff-time,
+              // then commit is live.
+              commitMeta.getCommitTime().compareTo(cutOffTimestamp) >= 0);
+    }
+  }
+
+  protected @NotNull IdentifiedResult computeExpiredContents(
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap, Reference reference) {
+    try (NessieApiV1 api = getApi(config)) {
+      return walkAllCommitsInReference(api, reference, liveContentsBloomFilterMap);
+    }
+  }
+
+  private int getTaskCount(List<Reference> references) {
+    int taskCount =
+        Integer.parseInt(
+            config.getOrDefault(
+                GCConfigConstants.CONF_NESSIE_GC_SPARK_TASK_COUNT,
+                String.valueOf(references.size())));
+    if (taskCount <= 0) {
+      LOGGER.warn(
+          "Invalid {} configuration: {} using default",
+          GCConfigConstants.CONF_NESSIE_GC_SPARK_TASK_COUNT,
+          taskCount);
+      taskCount = references.size();
+    }
+    return taskCount;
+  }
+
+  private Map<String, ContentBloomFilter> mergeLiveContentResults(
+      List<Map<String, ContentBloomFilter>> bloomFilterMaps) {
+    Map<String, ContentBloomFilter> output = new HashMap<>();
+    bloomFilterMaps.forEach(
+        map ->
+            map.forEach(
+                (k, v) -> {
+                  if (output.containsKey(k)) {
+                    output.get(k).merge(v);
+                  } else {
+                    output.put(k, v);
+                  }
+                }));
+    return output;
+  }
+
+  private Map<String, ContentBloomFilter> walkLiveCommitsInReference(
+      NessieApiV1 api,
+      Reference reference,
+      boolean isRefDroppedAfterCutoffTimeStamp,
+      Predicate<CommitMeta> liveCommitPredicate) {
+    Map<String, ContentBloomFilter> bloomFilterMap = new HashMap<>();
+    Set<ContentKey> liveContentKeys = new HashSet<>();
+    try (Stream<LogResponse.LogEntry> commits =
+        StreamingUtil.getCommitLogStream(
+            api, Detached.REF_NAME, reference.getHash(), null, null, OptionalInt.empty(), true)) {
+      ReferenceMetadata defaultRefMetadata =
+          api.getReference()
+              .refName(api.getDefaultBranch().getName())
+              .fetch(FetchOption.ALL)
+              .get()
+              .getMetadata();
+      long totalCommitsInDefaultReference =
```

Comment:
Calculating this value here is dangerous and will lead to BFs with different sizes that are in turn not mergeable.
The BF sizing parameters must be the same in every spark task.

Best would be to move the code that runs as a separate spark task to a separate class, so it's more obvious what's running where and when it gets initialized.

Reply:
only public methods (just two) in `DistributedIdentifyContents` happens in spark driver, rest all happens in executor. It is hard to separate. So I think it is ok to keep in same file. 

I now moved `totalCommitsInDefaultReference` calculation to driver and passing this value to executor.